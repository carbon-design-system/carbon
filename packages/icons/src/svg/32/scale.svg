<svg id="icon" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 32 32"><defs><style>.cls-1{fill:none;}</style></defs><title>scale</title><path d="M13,17H7a2,2,0,0,0-2,2v6a2,2,0,0,0,2,2h6a2,2,0,0,0,2-2V19A2,2,0,0,0,13,17ZM7,25V19h6v6Z"/><path d="M19,21v2h6a2,2,0,0,0,2-2V7a2,2,0,0,0-2-2H11A2,2,0,0,0,9,7v6h2V7H25V21"/><rect id="_Transparent_Rectangle_" data-name="&lt;Transparent Rectangle&gt;" class="cls-1" width="32" height="32"/></svg>