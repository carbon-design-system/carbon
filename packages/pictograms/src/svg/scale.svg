<?xml version="1.0" encoding="iso-8859-1"?>
<!-- Generator: Adobe Illustrator 24.1.2, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 viewBox="0 0 32 32" style="enable-background:new 0 0 32 32;" xml:space="preserve">
<path id="scale_1_" d="M31,31.36H1c-0.199,0-0.36-0.161-0.36-0.36V1c0-0.199,0.161-0.36,0.36-0.36h30c0.199,0,0.36,0.161,0.36,0.36
	v30C31.36,31.199,31.199,31.36,31,31.36z M1.36,30.64h29.28V1.36H1.36V30.64z M11.066,14.453c-0.092,0-0.184-0.035-0.254-0.105
	L5.12,8.656c-0.141-0.141-0.141-0.368,0-0.509C8.025,5.241,11.889,3.64,15.999,3.64c4.109,0,7.974,1.601,10.88,4.507
	c0.067,0.067,0.105,0.159,0.105,0.254s-0.038,0.187-0.105,0.254l-5.692,5.692c-0.067,0.068-0.159,0.105-0.255,0.105l0,0
	c-0.096,0-0.188-0.038-0.255-0.105c-1.249-1.25-2.91-1.938-4.678-1.938c-1.767,0-3.428,0.688-4.678,1.938
	C11.25,14.418,11.159,14.453,11.066,14.453z M5.887,8.404l5.186,5.186c1.264-1.148,2.862-1.81,4.567-1.892V8.056h0.72v3.642
	c1.705,0.082,3.304,0.744,4.566,1.892l5.187-5.186c-1.045-0.996-2.212-1.82-3.466-2.459l-0.834,1.444l-0.623-0.36l0.806-1.395
	c-1.755-0.787-3.662-1.222-5.635-1.269V6h-0.72V4.364c-1.974,0.047-3.882,0.482-5.637,1.269l0.808,1.399l-0.624,0.36L9.352,5.944
	C8.098,6.583,6.931,7.408,5.887,8.404z"/>
<rect id="_Transparent_Rectangle" style="fill:none;" width="32" height="32"/>
</svg>
