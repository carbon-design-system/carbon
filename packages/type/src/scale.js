/**
 * Copyright IBM Corp. 2018, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

/**
 * Get the type size for the given step
 * @param {number} step
 * @returns {number}
 */
export function getTypeSize(step) {
  if (step <= 1) {
    return 12;
  }
  // Yn = Yn-1 + {FLOOR[(n - 2) / 4] + 1} * 2
  return getTypeSize(step - 1) + Math.floor((step - 2) / 4 + 1) * 2;
}

/**
 * The default type scale for 23 steps. Inlined as an array here through running
 * the follow step:
 *
 * > Array.from({ length: 23 }, (_, i) => getTypeSize(i + 1))
 */
export const scale = [
  12, 14, 16, 18, 20, 24, 28, 32, 36, 42, 48, 54, 60, 68, 76, 84, 92, 102, 112,
  122, 132, 144, 156,
];
