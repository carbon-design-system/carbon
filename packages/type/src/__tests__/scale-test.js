/**
 * Copyright IBM Corp. 2018, 2023
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @jest-environment node
 */

import { getTypeSize, scale } from '../scale';

describe('scale', () => {
  it('should export the type scale', () => {
    expect(scale).toMatchSnapshot();
  });

  describe('getTypeSize', () => {
    it('should return the base font for steps <= 1', () => {
      expect(getTypeSize(1)).toBe(12);
      expect(getTypeSize(0)).toBe(12);
    });

    it('should support steps greater than 1', () => {
      expect(() => {
        getTypeSize(2);
        getTypeSize(10);
        getTypeSize(20);
      }).not.toThrow();
    });
  });
});
