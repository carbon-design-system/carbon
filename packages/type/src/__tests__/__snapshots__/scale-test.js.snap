// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scale should export the type scale 1`] = `
[
  12,
  14,
  16,
  18,
  20,
  24,
  28,
  32,
  36,
  42,
  48,
  54,
  60,
  68,
  76,
  84,
  92,
  102,
  112,
  122,
  132,
  144,
  156,
]
`;
